Source/Core/S2Cell.js

```diff
@@ -9,6 +9,90 @@ import Ellipsoid from "./Ellipsoid.js";
+ * --------
+ * Each cell in S2 can be uniquely identified using a 64-bit unsigned integer. The first 3 bits of the cell ID are the face bits, i.e.
+ * they indicate which of the 6 faces of the cube the cell lies on. After the first bits are the position bits, i.e. they indicate the position
+ * of the cell along the Hilbert curve. After the positions bits is the sentinel bit, which is always set to 1, and it indicates the level of the
+ * cell. Again, the level can be between 0 and 30 in S2.
+ *
+ *   Cell ID (base 10): 3170534137668829184
+ *   Cell ID (base 2) : 0010110000000000000000000000000000000000000000000000000000000000
+ *
+ *   001 0110000000000000000000000000000000000000000000000000000000000
+ *   fff pps----------------------------------------------------------
+ *
+ * For the cell above, we can see that it lies on face 1, with a Hilbert index of 2.
+ *
+ * Cell Subdivision:
+ * ------------------
+ * Cells in S2 subdivide recursively using quadtree subdivision. For each cell, you can get a child of index [0-3]. To compute the child at index i,
+ * insert the base 2 representation of i to the right of the parent's position bits. Ensure that the sentinel bit is also shifted two places to the right.
+ *
+ *   Parent Cell ID (base 10) : 3170534137668829184
+ *   Parent Cell ID (base 2)  : 0010110000000000000000000000000000000000000000000000000000000000
+ *
+ *   001 0110000000000000000000000000000000000000000000000000000000000
+ *   fff pps----------------------------------------------------------
+ *
+ *   To get the 3rd child of the cell above, we insert the binary representation of 3 to the right of the parent's position bits:
+ *
+ *   001 0111100000000000000000000000000000000000000000000000000000000
+ *   fff pppps--------------------------------------------------------
+ *         ^^
+ *
+ *   Child(3) Cell ID (base 10) : 3386706919782612992
+ *   Child(3) Cell ID (base 2)  : 0010111100000000000000000000000000000000000000000000000000000000
+ *
+ * Cell Token:
+ * -----------
+ * To provide a more concise representation of the S2 cell ID, we can their hexadecimal representation.
+ *
+ *   Cell ID (base 10): 3170534137668829184
+ *   Cell ID (base 2) : 0010110000000000000000000000000000000000000000000000000000000000
+ *
+ *   We remove all trailing zero bits, until we reach the nybble (4 bit mulitple) that contains the sentinel bit.
+ *
+ *   0010110000000000000000000000000000000000000000000000000000000000
+ *   fffpps--XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+ *
+ *   We convert the remaining bits to their hexadecimal representation.
+ *
+ *   0010 1100
+ *    "2"  "c"
+ *
+ *   Cell Token: "2c"
+ *
+ * To compute the cell ID from the token, we simple add enough zeros to the right to make the ID span 64 bits.
+ *
+ * Coordinate Transforms:
+ * ----------------------
+ *
+ * To go from a cell in S2 to a point on the ellipsoid, the following order of transforms is applied:
+ *
+ *   1. (Cell ID): S2 cell ID
```

Comment:
Isn't there technically `(face, position, level)` in between cell ID and `(face I, J)`?

Reply:
Extracting that is part of the conversion to (Face, I, J). We don't have a dedicated function for that, since it's trivial to extract.